src/v/kafka/server/datalake_throttle_manager.cc

```diff
@@ -44,37 +45,37 @@ datalake_throttle_manager::status datalake_throttle_manager::status::operator+(
   const datalake_throttle_manager::status& o) const {
     return datalake_throttle_manager::status{
       .max_shares_assigned = max_shares_assigned || o.max_shares_assigned,
-      .overdue_translation_partition_count
-      = overdue_translation_partition_count
-        + o.overdue_translation_partition_count,
-      .partitions_translation_blocked = partitions_translation_blocked
-                                        + o.partitions_translation_blocked};
+      .total_translation_backlog = total_translation_backlog
+                                   + o.total_translation_backlog};
 }
 
-bool datalake_throttle_manager::status::needs_throttling() const {
-    return max_shares_assigned
-           && (overdue_translation_partition_count > 0 || partitions_translation_blocked > 0);
+bool datalake_throttle_manager::needs_throttling() const {
+    return _translation_status.max_shares_assigned
+           && _translation_status.total_translation_backlog
+                > _backlog_size_throttling_ratio() * _disk_space_info.total;
```

Comment:
how do we protect against the case where `_disk_space_info` is not initialized yet?